### PR TITLE
added draft media_play implementation

### PR DESCRIPTION
**Pull request by shivgarg**

### CODE REVIEW

**Comment by shivgarg** [comment]
I did not check compilation, please review the structure and then i will do the cosmetic changes to make it production ready

File: framework/include/media/media_player.h [hunk 1]
```diff
@@ -33,7 +33,14 @@ typedef enum {
...
 	PLAY_DRAINING
-} play_state_e;
+} play_state_t;
+
+typedef enum {
+	PLAY_OK,
+	PLAY_ERROR_THREAD_INIT,
+	PLAY_ERROR_MALLOC,
+	PLAY_ERROR_DRIVER
+} play_result_t;
```
Comment (jarvisjang) [inline framework/include/media/media_player.h]: According to the coding scheme, enum must be ended of _e.

File: framework/src/media/media_player.c [hunk 2]
```diff
@@ -21,20 +21,63 @@
...
 	int fd;
-	play_state_e state;	
+	play_state_t state;	
 	media_format_e format;
 };
 
 static pthread_mutexattr_t g_mutex_attr_play;
 static pthread_mutex_t g_mutex_play;
-static struct media_player_s g_player;
+static struct media_player_s *g_player;
```
Comment (jarvisjang) [inline framework/src/media/media_player.c]: I think g_player doesn't need to be a pointer. Do you have any reason that g_player is a pointer?

File: framework/src/media/media_player.c [hunk 3]
```diff
@@ -21,20 +21,63 @@
...
+static play_state_t g_state = PLAY_ERROR;
```
Comment (jarvisjang) [inline framework/src/media/media_player.c]: g_player has play_state_t. Is this different from g_player.state?

File: framework/include/media/media_player.h (hunk 1 above)
Comment (shivgarg) [reply framework/include/media/media_player.h]: In coding rule all typedef variables , end with _t.

File: framework/src/media/media_player.c (hunk 2 above)
Comment (shivgarg) [reply framework/src/media/media_player.c]: Ok i will change

File: framework/src/media/media_player.c (hunk 3 above)
Comment (shivgarg) [reply framework/src/media/media_player.c]: currently both of these are same, but i thought g_state represents state of the player and g_player.state represents the state of that particular play object. I think it is better to separate these two now, even though they are same as in future we may need both the variables.

**Pull request merged**